client.go

```diff
@@ -122,7 +123,7 @@ func NewClientPipe(rd io.Reader, wr io.WriteCloser, opts ...ClientOption) (*Clie
 				WriteCloser: wr,
 			},
 			inflight: make(map[uint32]chan<- result),
-			errCh:    make(chan error, 1),
+			errCond:  sync.NewCond(new(sync.Mutex)),
```

Comment:
Since this triggers at most only once, rather than a `sync.Cond` you can also use the feature that a receive on a closed channel immediately sends the zero value.

So, here you would have a `closed: make(chan struct{})`, where you have the whole `c.errCond.Wait()` block, all of that can be just `<-c.closed; return c.err`. Then instead of `Broadcast` you just have `close(c.closed)`.

This tends to be a much more simple way to get a single-use state-change trigger of a broadcast unlock. While `sync.Cond` is better if it needs to broadcast repeated state-changes.